#ifndef SERVERPANELMAINWINDOW_H

///////////////////////////////////////////////////////////////////////////////
/// Definitions //////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////

#define SERVERPANELMAINWINDOW_H

///////////////////////////////////////////////////////////////////////////////
/// Headers //////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////

#include <QWidget>
#include <stdlib.h>
#include <iostream>
#include <QBuffer>
#include <QtCrypto/QtCrypto>
#include <QtGui/QMessageBox>
#include "Json.h"

///////////////////////////////////////////////////////////////////////////////
/// Namespace ////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////

namespace Ui {
    // Add our main window class to the namespace
    class ServerPanelMainWindow;
}

///////////////////////////////////////////////////////////////////////////////
/// ServerPanelMainWindow Definition /////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////

class ServerPanelMainWindow : public QWidget {
    // Make sure we load the QObject
    Q_OBJECT
// Public
public :
    // Singleton
    static ServerPanelMainWindow* Instance();
    // Destructor
    ~ServerPanelMainWindow();
// Private
private:
    // Constructor
    explicit ServerPanelMainWindow(QWidget *cParent = 0);
    // Properties
    static ServerPanelMainWindow* mInstance;      // Singleton Instance
    Ui::ServerPanelMainWindow*    mUserInterface; // User Interface Instance
// Private slots
private slots :
    void TrySignIn();
};

#endif // SERVERPANELMAINWINDOW_H
